### PR TITLE
[WIP] docs(samples): add Dart version of user-input example

### PR DESCRIPTION
**Pull request by tomplusplus**
@kevmoo , @kwalrath . Please Review. This is pretty much a direct copy from what exist in the jade files except with the most current version of angular.

Thanks.

### CODE REVIEW

File: public/docs/_examples/user-input/dart/lib/todo_list.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+class TodoList {
+  List<String> todos = [
+    'Eat breakfast',
+    'Walk dog',
+    'Breathe',
+    'Learn Angular'
+  ];
+
+  addTodo(String todo) {
+    window.console.log(todos.indexOf(todo));
```
Comment (kwalrath) [inline public/docs/_examples/user-input/dart/lib/todo_list.dart]: Remove this line. (If we kept it, it would be "print" instead of window.console.log().)

**Comment by kwalrath** [comment]
lgtm, after you fix/remove window.console.log. (If you've used that in any other samples, we should change it to print().)

**Comment by kwalrath** [comment]
One more thing I forgot: Please add a comment with the text `#docregion` at the top of every file, using whatever the comment syntax is for that type of file. For example, in a Dart file:

```
// #docregion
...
```

More details at https://angular.io/docs/styleguide.html. (I'll take care of the forms examples, since I'm working on that guide right now.)

Could you also remove any filenames you see at the tops of files? Those aren't necessary any more with the new styles.

Thanks!!

File: public/docs/_examples/user-input/dart/lib/todo_list.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
+library user_input.todo_list;
+
+import 'dart:html';
+
+import 'package:angular2/angular2.dart';
+
+@Component(selector: 'todo-list')
+@View(
```
Comment (kwalrath) [inline public/docs/_examples/user-input/dart/lib/todo_list.dart]: I think you can drop @View and put the template in @Component.

File: public/docs/_examples/user-input/dart/lib/todo_list.dart (hunk 1 above)
Comment (tomplusplus) [inline public/docs/_examples/user-input/dart/lib/todo_list.dart]: Oh yeah - that snuck in there I meant to remove it.

File: public/docs/_examples/user-input/dart/lib/todo_list.dart (hunk 2 above)
Comment (tomplusplus) [inline public/docs/_examples/user-input/dart/lib/todo_list.dart]: Ok good to know. I've seen both ways through out other examples so i wasn't sure if we wanted to unify on one way over the other for all examples, or if we wanted to show different ways in different parts.

Comment (kwalrath) [inline public/docs/_examples/user-input/dart/lib/todo_list.dart]: Yep, it started out with @View being required, but now @View shouldn't be required unless you have multiple views (which might not even be possible yet).

Comment (tomplusplus) [inline public/docs/_examples/user-input/dart/lib/todo_list.dart]: I'm using @View in the other samples (pipes and forms) that has already been merged. Should i update those and do a new PR ?

Comment (kwalrath) [inline public/docs/_examples/user-input/dart/lib/todo_list.dart]: I'll fix forms, but do please fix pipes. Thanks!

**Pull request closed without merging**